### PR TITLE
Castling

### PR DESCRIPTION
**Pull request by jiro-sains**
extended valid_move for king - added castling method.

verifies king hasn't moved before
verifies rook on side hasn't moved before
checks there are no pieces in between

### CODE REVIEW

File: app/models/king.rb [hunk 1]
```diff
@@ -1,9 +1,37 @@
+require 'pry'
```
Comment (apolishch) [inline app/models/king.rb]: try and remember to take superfluous debugging gems out before merges to master

**Review by apolishch — DISMISSED** [review]

File: app/models/king.rb [hunk 2]
```diff
@@ -1,9 +1,37 @@
...
   end
-end
\ No newline at end of file
+
+  def castling?(destination_x, destination_y)
+   !self.has_moved? && !rook_on_side_moved?(destination_x, destination_y) &&
+   		!pieces_between?(destination_x, destination_y)
+  end
+
+  def rook_on_side_moved?(destination_x, destination_y)
```
Comment (bountonw) [inline app/models/king.rb]: Are there any other rooks besides those that are on the side? Why not just name it `rook_moved?`

File: spec/models/king_spec.rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
+require 'pry'
```
Comment (bountonw) [inline spec/models/king_spec.rb]: Remove as per Abe's suggestion

File: app/models/king.rb (hunk 2 above)
Comment (bountonw) [inline app/models/king.rb]: Aren't all rooks on the side? Why not call it `rook_moved?`

**Review by bountonw — DISMISSED** [review]
Very minor change suggestions.

**Review by bountonw — APPROVED** [review]

**Pull request merged**